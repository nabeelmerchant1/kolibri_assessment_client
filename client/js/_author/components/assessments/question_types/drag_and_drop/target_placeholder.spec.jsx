import React              from 'react';
import { shallow }        from 'enzyme';
import TargetPlaceholder  from './target_placeholder';

describe('target_placeholder component', () => {
  let props;
  let result;
  let changer;

  beforeEach(() => {
    changer = false;
    props = {
      onClick: () => { changer = true; },
    };
    result = shallow(<TargetPlaceholder {...props} />);
  });

  it('renders the component as a snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('div is clicked and changes value', () => {
    result.find('.au-c-drag-and-drop__target-image-placeholder').simulate('click');
    expect(changer).toBeTruthy();
  });
});
