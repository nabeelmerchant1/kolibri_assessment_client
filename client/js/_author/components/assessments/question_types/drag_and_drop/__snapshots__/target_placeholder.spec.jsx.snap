// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`target_placeholder component renders the component as a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="au-c-drag-and-drop__target-image-placeholder"
    onClick={[Function]}
>
    Add Target Image
</div>,
  "nodes": Array [
    <div
      className="au-c-drag-and-drop__target-image-placeholder"
      onClick={[Function]}
>
      Add Target Image
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <targetPlaceholder
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="au-c-drag-and-drop__target-image-placeholder"
          onClick={[Function]}
>
          Add Target Image
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="au-c-drag-and-drop__target-image-placeholder"
          onClick={[Function]}
>
          Add Target Image
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <targetPlaceholder
    onClick={[Function]}
/>,
}
`;
